fhir_pyrate/ahoy.py

```diff
@@ -39,19 +39,21 @@ def __init__(
         username: str = None,
         auth_method: Optional[str] = "password",
         token: str = None,
-        token_refresh_time_minutes: int = 15,
+        max_login_attempts: int = 5,
     ) -> None:
         self.auth_type = auth_type
         self.auth_method = auth_method
         self.auth_url = auth_url
         self.refresh_url = refresh_url
         self.username = username
-        self.token_refresh_minutes = token_refresh_time_minutes
         self._user_env_name = "FHIR_USER"
         self._pass_env_name = "FHIR_PASSWORD"
         self.token = token
         self.session = requests.Session()
         self.auth_time = None
+        self.login_reattempted_times = 0
+        self.max_login_attempts = max_login_attempts
+        self.session.hooks["response"].append(self._refresh_hook)
```

Comment:
I think this might interfere with other auth schemes. The best option would be to add this kind of refresh mechanism as a separate class derived from `requests.auth.AuthBase`. In this derived class the response hook could be added.